lighthouse-core/test/audits/treemap-data-test.js

```diff
@@ -0,0 +1,335 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const TreemapData_ = require('../../audits/treemap-data.js');
+const networkRecordsToDevtoolsLog = require('../network-records-to-devtools-log.js');
+const {loadSourceMapAndUsageFixture, makeParamsOptional} = require('../test-utils.js');
+
+/* eslint-env jest */
+
+const TreemapData = {
+  audit: makeParamsOptional(TreemapData_.audit),
+  prepareTreemapNodes: makeParamsOptional(TreemapData_.prepareTreemapNodes),
+};
+
+/**
+ * @param {string} url
+ * @param {number} resourceSize
+ * @param {LH.Crdp.Network.ResourceType} resourceType
+ */
+function generateRecord(url, resourceSize, resourceType) {
+  return {url, resourceSize, resourceType};
+}
+
+describe('TreemapData audit', () => {
+  describe('squoosh fixture', () => {
+    /** @type {import('../../audits/treemap-data.js').TreemapData} */
+    let treemapData;
+    beforeAll(async () => {
+      const context = {computedCache: new Map()};
+      const {map, content, usage} = loadSourceMapAndUsageFixture('squoosh');
+      const mainUrl = 'https://squoosh.app';
+      const scriptUrl = 'https://squoosh.app/main-app.js';
+      const networkRecords = [generateRecord(scriptUrl, content.length, 'Script')];
+
+      // Add a script with no source map or usage.
+      const noSourceMapScript = {src: 'https://sqoosh.app/no-map-or-usage.js', content: '// hi'};
+      networkRecords.push(
+        generateRecord(noSourceMapScript.src, noSourceMapScript.content.length, 'Script')
+      );
+
+      const artifacts = {
+        URL: {requestedUrl: mainUrl, finalUrl: mainUrl},
+        JsUsage: {[usage.url]: [usage]},
+        devtoolsLogs: {defaultPass: networkRecordsToDevtoolsLog(networkRecords)},
+        SourceMaps: [{scriptUrl: scriptUrl, map}],
+        ScriptElements: [{src: scriptUrl, content}, noSourceMapScript],
+      };
+      const results = await TreemapData.audit(artifacts, context);
+
+      // @ts-expect-error: Debug data.
+      treemapData = results.details.treemapData;
+    });
+
+    it('basics', () => {
+      expect(Object.keys(treemapData)).toEqual(['scripts', 'resources']);
+    });
+
+    it('scripts', () => {
```

Comment:
at least for me, test descriptions often go a long way to give an idea of what the test (and the original author of the test) was after, so I'm not starting at square 1 when a test fails and I have to figure out if I broke something or the test needs to be updated. OTOH, sometimes descriptions are just obvious statements, but either way it doesn't take much to turn these into sentence fragments and at least keep our tests consistent in style :)